build-logic/binary-compatibility/src/main/groovy/gradlebuild.binary-compatibility.gradle

```diff
@@ -136,14 +136,14 @@ def checkBinaryCompatibility = tasks.register("checkBinaryCompatibility", Japicm
     onlyModified = false
     failOnModification = false // we rely on the custom report to fail or not
     ignoreMissingClasses = true // because of a missing scala.runtime.AbstractFunction0 class
-    richReport {
-        includedClasses = toPatterns(PublicApi.includes + PublicKotlinDslApi.includes)
-        excludedClasses = toPatterns(PublicApi.excludes + PublicKotlinDslApi.excludes)
-
-        title = "Binary compatibility report for Gradle ${isSnapshot.get() ? "${baseVersion.get()}-SNAPSHOT" : version} since ${compatibilityBaselineVersion}"
-        destinationDir = file("$buildDir/reports/binary-compatibility")
-        reportName = "report.html"
-        description = """
+    richReport({
```

Comment:
Would extracting a lambda parameter such as `report` cause the compiler to know this is an Action, not Closure, and allow you to remove the cast?

Reply:
I think an extra lambda parameter is also a valid closure?

```
richReport({ it ->
    ...
} as Closure)
```

Reply:
There's no `Closure` variant here. The only one is `Action`. I think this is a mistake on the plugin's part.

AFAIK, this is the only way to generate an `Action`. And it now uses `it` because Groovy doesn't do what Kotlin does with the `Action` interface and set the receiver to the parameter. I would've used `ConfigureUtil` here which makes more sense, but I was unsure if I should use an internal API like that, even in our scripts.